1-development/list-posts-by-rating/README.md

```diff
@@ -0,0 +1,78 @@
+# list-posts-by-rating Submission
+
+My submition for the requested InvoicExpress Challenge.
+
+## Tech Stack
+
+This solution uses Postgresql for its database.
+
+The application is programed in JAVA, with the Spring Boot framework.
+
+## Usage
+Being in the /list-posts-by-rating/ directory:
+```bash
+psql -U postgres --file listpostsbyrating_db.sql
+```
+- this creates the database and the "posts" table. It also populates the database with a few posts.
+
+
+```bash
+mvn clean install
+```
+- this builds Spring Boot Project with Maven (clean command just to be sure)
+
+```bash
+mvn spring-boot:run
+```
+- runs the Spring Boot app using Maven
+
+ 
+Now the server is running. Go to localhost on a web browser and make the requests to the API.
+```bash
+http://localhost:8080/posts
+```
+- this returns a response with all the posts currently present in the database, ordered by upvote ratio descending, downvote ratio ascending and total votes descending. 
+
+```bash
+http://localhost:8080/upvote/:{post_id}
+```
+- this increments the number of upvotes by 1 on the post present in the database with {post_id}
+
+```bash
+http://localhost:8080/downvote/:{post_id}
+```
+- this increments the number of downvotes by 1 on the post present in the database with {post_id}
+
+### Extra
+
+I also programmed an extra POST Request to add a post, for debugging purposes.
+```bash
+http://localhost:8080/addpost

```

Comment:
Why do you have an extra endpoint for this matter? What other option could you choose?

Reply:
I could just add the posts directly to the database and then make the GET request to retrieve the posts, but because I never really had programmed an API from scratch in JAVA, I decided to start by doing this POST request, just to get the hang of it. And then I decided to leave it on the submission anyway.